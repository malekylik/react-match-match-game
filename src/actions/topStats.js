import {
    SET_TOP_STATS,
    SET_TOP_STATS_LOADING,
} from '../constants/actions';
import { convertGetData } from '../utils';
import { getFromDatabase } from '../utils/api';

export const setTopStats = (topStats) => ({
    type: SET_TOP_STATS,
    payload: topStats,
});

export const setTopStatsLoading = (isLoading) => ({
    type: SET_TOP_STATS_LOADING,
    payload: isLoading,
});

export const loadTop = () => {
    return async (dispatch) => {
        dispatch(setTopStatsLoading(true));
        const top = (await getFromDatabase()).result.sort((l, r) => (r.score - l.score));

        dispatch(setTopStats(convertGetData(top)));
        dispatch(setTopStatsLoading(false));
    };
};
