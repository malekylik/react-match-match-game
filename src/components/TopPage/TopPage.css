.top {
    text-shadow: 0 1px black;
    background-color: rgba(5,5,5,.95);
    border-radius: 20px;
    z-index: 10;
    position: absolute;
    left: 5px;
    top: 55px;
    overflow-y: scroll;
    max-height: 590px;
}

.top__table {
    font-size: 1.8rem;
    color: white;
}

.top__th {
    text-align: center;
    padding: 15px 10px;
}
