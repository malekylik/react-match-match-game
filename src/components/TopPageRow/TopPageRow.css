.top__td {
    text-align: center;
    padding: 5px 10px;
    padding: 15px 10px;
}
