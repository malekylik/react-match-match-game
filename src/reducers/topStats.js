import { List, Map } from 'immutable';

import {
    SET_TOP_STATS,
    SET_TOP_STATS_LOADING,
} from '../constants/actions';

export const initialState = Map({
    topStats: List(),
    isLoading: false,
});

const topStats = (state = initialState, action) => {
    switch (action.type) {
        case SET_TOP_STATS:
            return state.set('topStats', List(action.payload));
        case SET_TOP_STATS_LOADING:
            return state.set('isLoading', action.payload);
        default: return state;
    }
};

export const getTopStats = state => state.get('topStats');
export const getIsTopStatsLoading = state => state.get('isLoading');

export default topStats;
