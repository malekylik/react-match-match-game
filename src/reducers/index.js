import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';

import playerInfo from './playerInfo';
import player from './player';
import gameInfo from './gameInfo';
import topStats from './topStats';

export default combineReducers({
    playerInfo,
    gameInfo,
    player,
    topStats,
    routing: routerReducer,
});
