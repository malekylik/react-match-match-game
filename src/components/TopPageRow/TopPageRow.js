import React from 'react';
import PropTypes from 'prop-types';

import './TopPageRow.css';

const TopPageRow = ({ number, email, name, score }) => (
    <tr>
        <td className="top__td">
            {number}
        </td>
        <td className="top__td">
            {email}
        </td>
        <td className="top__td">
            {name}
        </td>
        <td className="top__td">
            {score}
        </td>
    </tr>
);

TopPageRow.propTypes = {
    number: PropTypes.number.isRequired, 
    email: PropTypes.string, 
    name: PropTypes.string, 
    score: PropTypes.number, 
};

export default TopPageRow; 
