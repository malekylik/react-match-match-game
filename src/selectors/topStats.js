import React from 'react';

import { createSelector } from 'reselect';

import TopPageRow from '../components/TopPageRow/TopPageRow';

import {
    getTopStats as _getTopStats,
    getIsTopStatsLoading as _getIsTopStatsLoading,
} from '../reducers/topStats';

export const getTopStats = state => _getTopStats(state.topStats);
export const getTopStatsAsTopPageRows = createSelector(getTopStats,
    topStats => topStats.map((rows, i) => (<TopPageRow key={i + 1} number={i + 1} {...rows} />)).toArray());
export const getIsTopStatsLoading = state => _getIsTopStatsLoading(state.topStats);
