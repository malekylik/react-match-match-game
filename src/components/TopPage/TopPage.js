import React from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';

import { getTopStatsAsTopPageRows } from '../../selectors/topStats';

import './TopPage.css';

const TopPage = ({ topStatsAsTopPageRows }) => {
    return (
        <div className="top">
            <table className="top__table">
                <tbody>
                    <tr>
                        <th className="top__th">
                            â„–
                    </th>
                        <th className="top__th">
                            email
                    </th>
                        <th className="top__th">
                            Name
                    </th>
                        <th className="top__th">
                            Score
                    </th>
                    </tr>
                    {topStatsAsTopPageRows}
                </tbody>
            </table>
        </div>
    );
}

TopPage.propTypes = {
    topStatsAsTopPageRows: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element)
    ]),
};

export default connect((state) => ({
    topStatsAsTopPageRows: getTopStatsAsTopPageRows(state),
}))(TopPage); 
