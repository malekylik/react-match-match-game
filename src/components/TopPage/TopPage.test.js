import React from 'react';
import configureStore from 'redux-mock-store';

import { shallow } from 'enzyme';

import TopPage from './TopPage';
import TopPageRow from '../TopPageRow/TopPageRow';

import { initialState as topStatsInitialState } from '../../reducers/topStats';

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('TopPage', () => {
    it('Should match snapshot with data', () => {
        const initialState = {
            topStats: topStatsInitialState,
        };

        const store = mockStore(initialState);

        const topPageRowsData = [
            {
                email: 'test1@test1.test',
                name: 'name1',
                score: 1,
            },
            {
                email: 'test2@test2.test',
                name: 'name2',
                score: 2,
            },
        ];

        const wrapper = shallow((<TopPage store={store} />));

        const topStatsAsTopPageRows = topPageRowsData.map((rows, i) => (<TopPageRow key={i + 1} number={i + 1} {...rows} />));
        wrapper.setProps({ topStatsAsTopPageRows });

        expect(wrapper).toMatchSnapshot();
    });
});
