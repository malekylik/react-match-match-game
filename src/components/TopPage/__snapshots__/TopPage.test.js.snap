// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPage Should match snapshot with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TopPage)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    topStatsAsTopPageRows={
      Array [
        <TopPageRow
          email="test1@test1.test"
          name="name1"
          number={1}
          score={1}
        />,
        <TopPageRow
          email="test2@test2.test"
          name="name2"
          number={2}
          score={2}
        />,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dispatch": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "topStatsAsTopPageRows": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dispatch": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "topStatsAsTopPageRows": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
